src/Core/Billing/Services/Implementations/SubscriberService.cs

```diff
@@ -609,25 +610,54 @@ public async Task UpdateTaxInformation(
             }
         });
 
-        if (!subscriber.IsUser())
+        var taxId = customer.TaxIds?.FirstOrDefault();
```

Comment:
Non-blocker: consider moving this into its own method to improve readability and testability.